src/pydicom/pixels/utils.py

```diff
@@ -223,6 +231,339 @@ def as_pixel_options(ds: "Dataset", **kwargs: Any) -> dict[str, Any]:
     return opts
 
 
+def compress(
+    ds: "Dataset",
+    transfer_syntax_uid: str,
+    arr: "np.ndarray | None" = None,
+    *,
+    encoding_plugin: str = "",
+    encapsulate_ext: bool = False,
+    jls_error: int | None = None,
+    j2k_cr: list[float] | None = None,
+    j2k_psnr: list[float] | None = None,
+    **kwargs: Any,
+) -> "Dataset":
+    """Compress uncompressed pixel data and update `ds` in-place with the
+    resulting :dcm:`encapsulated<part05/sect_A.4.html>` codestream.
+
+    .. versionadded:: 3.0
+
+    The dataset `ds` must already have the following
+    :dcm:`Image Pixel<part03/sect_C.7.6.3.html>` module elements present
+    with correct values that correspond to the resulting compressed
+    pixel data:
+
+    * (0028,0002) *Samples per Pixel*
+    * (0028,0004) *Photometric Interpretation*
+    * (0028,0008) *Number of Frames* (if more than 1 frame will be present)
+    * (0028,0010) *Rows*
+    * (0028,0011) *Columns*
+    * (0028,0100) *Bits Allocated*
+    * (0028,0101) *Bits Stored*
+    * (0028,0103) *Pixel Representation*
+
+    If *Samples per Pixel* is greater than 1 then the following element
+    is also required:
+
+    * (0028,0006) *Planar Configuration*
+
+    This method will add the file meta dataset if none is present and add
+    or modify the following elements:
+
+    * (0002,0010) *Transfer Syntax UID*
+    * (7FE0,0010) *Pixel Data*
+
+    If the compressed pixel data is too large for encapsulation using a
+    basic offset table then an :dcm:`extended offset table
+    <part03/sect_C.7.6.3.html>` will also be used, in which case the
+    following elements will also be added:
+
+    * (7FE0,0001) *Extended Offset Table*
+    * (7FE0,0002) *Extended Offset Table Lengths*
```

Comment:
I wonder if this needs a new SOP Instance UID, as it may change the data... probably not, I'm just not completely sure about it.

Reply:
Hmm, the standard does say that it needs new Instance UIDs when converting like this. I think probably I should add it to `compress()` and `decompress()` as a kwarg, defaulting to adding a new one to be conformant.